sc/shared/src/main/scala/sigma/compiler/phases/SigmaTyper.scala

```diff
@@ -178,9 +178,20 @@ class SigmaTyper(val builder: SigmaBuilder,
           error(s"Cannot get field '$n' in in the object $newObj of non-product type ${newObj.tpe}", sel.sourceContext)
       }
 
-    case app @ Apply(sel @ Select(obj, n, _), args) =>
-      val newSel = assignType(env, sel)
+    case app @ Apply(selOriginal @ Select(obj, nOriginal, resType), args) =>
       val newArgs = args.map(assignType(env, _))
+
+      // hack to make possible to write g.exp(ubi) for both unsigned and signed big integers
+      // could be useful for other use cases where the same front-end code could be
+      // translated to different methods under the hood, based on argument types
+      // todo: consider better place for it
+      val (n, sel) = if (nOriginal == "exp" && newArgs(0).tpe.isInstanceOf[SUnsignedBigInt.type]) {
+        val newName = "expUnsigned"
+        (newName, Select(obj, newName, resType))
+      } else {
+        (nOriginal, selOriginal)
+      }
```

Comment:
Hm, why not just use different method names? This will be the only place with such exception, so will lead to more questions than answers.

Reply:
The idea is to provide unified interface to users (ErgoScript) developers, so kinda polymorphic exp method (which is then translated by typer into exp or expUnsigned, depends on argument)